File: mario-less/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height;

    do
    {
        height = get_int("Height: ");
    }
    while (height < 1);

    // using blank space to produce the stair
    // two for loops and one if in each loop
    for (int i = 0; i < height; i++)
    {
        for (int j = 0; j < height; j++)
        {
            if (height - j > i + 1)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }
        printf("\n");
    }
}

File: sentimental-mario-more/mario.py
from cs50 import get_int

while True:
    number = get_int("Height: ")
    if number <= 8 and number >= 1:
        break

for i in range(number):
    print(" " * (number - 1 - i), end='')
    print("#" * (i + 1), end='')
    print(" " * 2, end='')
    print("#" * (i + 1), end='')
    print("")  # 本质上就是换行了，如果里面写\n则换了两行
